velox/connectors/hive/storage_adapters/s3fs/tests/S3MetricsAggregatorTest.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/storage_adapters/s3fs/S3MetricsAggregator.h"
+#include <folly/init/Init.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+#include "velox/connectors/hive/storage_adapters/s3fs/S3Metrics.h"
+
+namespace facebook::velox::filesystems {
+namespace {
+
+class S3MetricsAggregatorTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    aggregator_ = S3MetricsAggregator::getInstance().get();
+    ASSERT_NE(aggregator_, nullptr)
+        << "Failed to get S3MetricsAggregator instance";
+    // Manually reset relevant metrics before each test using the metric names
+    // from S3Metrics.h
+    aggregator_->resetMetric(kMetricS3ActiveConnections);
+    aggregator_->resetMetric(kMetricS3StartedUploads);
+    aggregator_->resetMetric(kMetricS3FailedUploads);
+  }
+
+  void TearDown() override {
+    // Cleanup or reset any global state if necessary
+  }
+
+  S3MetricsAggregator* aggregator_;
+};
+
+TEST_F(S3MetricsAggregatorTest, IncrementAndRetrieveMetrics) {
+  // Ensure the metric starts at 0
+  EXPECT_EQ(aggregator_->getMetric(kMetricS3ActiveConnections), 0);
+
+  // Increment the metric and check the value
+  aggregator_->incrementMetric(kMetricS3ActiveConnections);
+  EXPECT_EQ(aggregator_->getMetric(kMetricS3ActiveConnections), 1);
+
+  // Increment the metric again and check the value
+  aggregator_->incrementMetric(kMetricS3ActiveConnections);
+  EXPECT_EQ(aggregator_->getMetric(kMetricS3ActiveConnections), 2);
+}
+
+TEST_F(S3MetricsAggregatorTest, ResetMetrics) {
```

Comment:
Rethink if we need ResetMetric API. If possible remove it.

Reply:
The resetMetric API is still needed to ensure unit tests start with empty metrics.